packages/react-native-test-renderer/src/renderer/index.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+import type {Element, ElementType} from 'react';
+
+import * as FabricUIManager from './FabricUIManagerMock';
+
+import {act} from 'react-test-renderer';
+import ReactFabric from 'react-native/Libraries/Renderer/shims/ReactFabric';
+
+type ReactNode = {
+  children: Array<ReactNode>,
+  props: {...},
+  viewName: string,
+  instanceHandle: {
+    memoizedProps: {testID: ?string, ...},
+    ...
+  },
+};
+
+type RootReactNode = $ReadOnlyArray<ReactNode>;
+
+type RenderResult = {
+  toJSON: () => string,
+  findByTestID: (testID: string) => ReactNode,
+};
+
+function buildRenderResult(rootNode: RootReactNode): RenderResult {
+  return {
+    toJSON: () => stringify(rootNode),
+    findByTestID: (testID: string) => findByTestID(rootNode, testID),
+  };
+}
+
+export function render(
+  element: Element<ElementType>,
+): RenderResult {
+  const manager = FabricUIManager.getFabricUIManager();
+  if (!manager) {
+    throw new Error('No FabricUIManager found');
+  }
+  const containerTag = Math.round(Math.random() * 1000000);
+  act(() => {
+    ReactFabric.render(element, containerTag, () => {}, true);
+  });
+
+  // $FlowFixMe
+  const root: RootReactNode = manager.getRoot(containerTag);
+
+  if (root == null) {
+    throw new Error('No root found for containerTag ' + containerTag);
+  }
+
+  return buildRenderResult(root);
+}
+
+export function fireEvent(node: ReactNode, eventName: 'press') {
```

Comment:
If you want to expose simulating multi-event interactions, they should involve delays between events like `touchstart` and `touchend`, and hence be async.